src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstancesLoaderImpl.scala

```diff
@@ -23,18 +25,22 @@ private[tracker] class InstancesLoaderImpl(
 
   override def load(): Future[InstanceTracker.InstancesBySpec] = async {
 
-    // Join instances with app or pod.
+    case object Tick
+    val progressReportTick = 10.seconds
+    var loadedInstancesCount = 0
+
+    // Join instances with app or pod spec.
     val coreInstances = await(
-      repo.ids().mapAsync(parallelism = config.internalInstanceTrackerNumParallelLoads()) { instanceId =>
-        async {
-          await(repo.get(instanceId)) match {
-            case None =>
-              await(expungeOrphanedInstance(instanceId)); None
-            case Some(coreInstance) => Some(coreInstance)
-          }
+      repo.ids().mapAsync(parallelism = config.internalInstanceTrackerNumParallelLoads())(loadInstance)
+        .map(Left(_))
```

Comment:
I think this would be better if we used alsoTo, instead. Then we wouldn't have to deal with the different Left / Right types

Reply:
Not sure `alsoTo` would work. We need one stream with instance IDs *and* ticks. How would you distinguish?

Reply:
Ping @timcharper